### PR TITLE
refactor activity_packs file

### PR DESCRIPTION
**Pull request by emilia-friedberg**
## WHAT
Big refactor of the `activity_packs` file, including rewriting as a functional component (including getting rid of unsafe lifecycle methods); breaking up large functions into smaller, more readable ones; using early returns instead of long if/else strings; and adding tests.

## WHY
I encountered this file recently on a totally separate project and found it a total nightmare to work with.

## HOW
I *should* have added the tests first, then confirmed they didn't change as I worked. Instead, I rewrote the code, checking in the browser as I worked that things still functioned as I expected them to, and then at the end wrote tests. I ran the same tests on develop, with the unchanged version of the file, and confirmed that the snapshot differences were non-issues (things like ids in library-imported components being different).

### Screenshots
(If applicable. Also, please censor any sensitive data)

### Notion Card Links
https://www.notion.so/quill/Clean-up-nightmare-file-b415e36d96ef41689822a973837e0d60?pvs=4

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? |  YES
Have you deployed to Staging?  YES
Self-Review: Have you done an initial self-review of the code below on Github? | YES
Spec Review: Have you reviewed the spec and ensured this meets requirements and/or matches design mockups? | YES

### CODE REVIEW

File: services/QuillLMS/client/app/bundles/Teacher/components/progress_reports/diagnostic_reports/activity_packs.jsx
Comment (eadams17) [inline services/QuillLMS/client/app/bundles/Teacher/components/progress_reports/diagnostic_reports/activity_packs.jsx:1]: not a big deal, but I was just curious why you didn't opt to make this a Typescript file?

**Review by eadams17 — APPROVED** [review]
This looks so much better! Thank you for cleaning this up

**Review by anathomical — APPROVED** [review]
Appreciate the clean-up here.  Sometimes I manage to forget how messy some of our older code is.

File: services/QuillLMS/client/app/bundles/Teacher/components/progress_reports/diagnostic_reports/activity_packs.jsx
Comment (emilia-friedberg) [reply services/QuillLMS/client/app/bundles/Teacher/components/progress_reports/diagnostic_reports/activity_packs.jsx:1]: @eadams17 haha, I totally should have! I was so intent on getting the bulk of the code itself cleaned up that I didn't even think about it. Noted for next time.

**Pull request merged**